packages/typescript/internal/ts_project.bzl

```diff
@@ -756,7 +778,28 @@ def ts_project_macro(
     if declaration_map:
         typing_maps_outs.extend(_out_paths(srcs, typings_out_dir, root_dir, allow_js, {"*": ".d.ts.map"}))
 
-    if not len(js_outs) and not len(typings_outs):
+    if type(transpiler) == "function" or type(transpiler) == "rule":
+        tsc_js_outs = []
+        tsc_map_outs = []
+        transpiler(
+            name = name + "_transpile",
+            srcs = srcs,
+            js_outs = js_outs,
+            map_outs = map_outs,
+            args = transpiler_args,
+            data = [],
+            tags = kwargs.get("tags", []),
+            visibility = kwargs.get("visibility", None),
```

Comment:
probably a few other well known attrs such as `test_only`. We have a list somewhere right?

Reply:
Yeah it's in aspect Bazel lib

Reply:
actually I think we're both confusing it with https://github.com/aspect-build/bazel-lib/blob/main/docs/utils.md#propagate_well_known_tags